fastats/linear_algebra/pinv.py

```diff
@@ -0,0 +1,27 @@
+
+import numpy as np
+
+
+def pinv(A):
```

Comment:
Why? `pinv` is supported by numba: https://numba.pydata.org/numba-doc/dev/reference/numpysupported.html#linear-algebra

Reply:
The same reason we have all the other functions that are also supported by numba - performance.

One of the main ideas of fastats is to move as much code into jittable python as possible, so that future compiler/hardware upgrades actually result in performance improvements. You don't get the same benefits from calling into compiled code, unless you recompile for a specific architecture.

```
from fastats.linear_algebra import pinv
import numpy as np
from numpy.linalg import pinv as npinv

arr2 = np.array([[1, 3], [2, 4]])

# Fastats version
%timeit pinv(arr2)
19.8 µs ± 267 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

# Numba/numpy version
%timeit npinv(arr2)
42.3 µs ± 2.5 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Obviously the performance profiles change for larger arrays, but the profiles above are indicative of what we've seen elsewhere when using a jitted linalg approach versus numba-wrapped numpy code.

Reply:
(updated all the above numbers with a jitted pinv profile, and changed the dtype to float as well)

Reply:
Obviously having a fastats version of matrix inverse will improve performance metrics even further :)

Reply:
I get all that, I just didn't quite expect such a performance gain. Wow, it's really worth it :) amazing.

Reply:
```
pinv_jit_fast = jit(nopython=True, parallel=True)(pinv_fast)
%timeit pinv_jit_fast(arr2)
1.17 µs ± 12 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

hehe
```

Reply:
Is this the PR function?

Reply:
I doubt it, @rjenc29 always adds a `_fast` suffix when he's done something special to speed it up 😄

Reply:
It's well worth spending the time and effort to speed these up - realistically having `pinv` reliant on `inv` is not the best idea, but speeding up both will have huge impacts elsewhere.

Reply:
Agreed - but given how much faster this version is, I'm eager to merge ASAP.